rclcpp/include/rclcpp/node.hpp

```diff
@@ -1039,6 +1039,9 @@ class Node : public std::enable_shared_from_this<Node>
    * rcl_interfaces::msg::SetParametersResult to indicate whether or not the
    * parameter should be set or not, and if not why.
    *
+   * The back returns a shared pointer to the callback handle.
+   * The callback is valid as long as the smart pointer is alive.
+   *
```

Comment:
I think that the doc is already there. is this not enough?

https://github.com/ros2/rclcpp/blob/b953bdddf8de213b4a051ecf2d668dad65ff9f89/rclcpp/include/rclcpp/node.hpp#L1097

Reply:
It's not the most obvious design so I figured that it might be good to make it _extra_ explicitly clear. It can be very confusing when the return value isn't persisted properly, as the user would just see the callback seemingly never get called. If they go put debug logging statements into rclcpp, the `post_set_callback_container_` mysteriously decreasing in size. Putting this detail at the beginning of the docstring might help with this.

 Taking a step back it would be more intuitive to tie the lifetime of the callback to the corresponding node to account for how most users won't scour scour the documentation and instead just assume the most likely behaviour. Then callbacks could be managed just with only the `add_{on, post}_set...` and `remove_{on, post}_set...` methods. Not sure if anyone's taking advantage of the behaviour of where the callback becomes invalidated when the smart pointer gets dropped.

Reply:
I guess the current design is nice if the object that registered the callback goes out of scope, then we probably want to unregistered the callback to avoid potential undefined behavior.